## Neural Network - Fashion-MNIST Dataset

# Import Tensorflow and Keras

In [17]:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten


## Load the Dataset - MNIST (MNIST = 28x28 grayscale images of handwritten digits (0–9))

In [18]:
from tensorflow.keras.datasets import mnist
(X_train,y_train),(X_test,y_test)= mnist.load_data()

x_train, x_test: Image pixel data

y_train, y_test: Corresponding labels (digit 0–9

## Normalize the data (We scale pixel values from [0–255] to [0–1]
Neural nets learn faster with small, consistent input)

In [19]:
x_train = x_train/255.0
x_train = x_test/255.0

##  Build the Model (Sequential)

In [21]:
model = keras.Sequential([
    layers.Flatten(input_shape=(28,28)), # converts 2D to 1D
    layers.Dense(128, activation='relu'), # hidden layer
    layers.Dropout(0.2), #Prevent overfitting
    layers.Dense(10,activation="softmax")
])

# Complie the Model

In [23]:
model.compile(optimizer ="adam",
              loss = 'sparse_categorical_crossentropy',
              metrics=['accuracy'])

# Train the model

In [24]:
model.fit(X_train,y_train, epochs=5)

Epoch 1/5


1875/1875 [==============================] - 6s 3ms/step - loss: 2.4572 - accuracy: 0.7460 
Epoch 2/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.6198 - accuracy: 0.8364
Epoch 3/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.4983 - accuracy: 0.8684
Epoch 4/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.4278 - accuracy: 0.8863
Epoch 5/5
1875/1875 [==============================] - 6s 3ms/step - loss: 0.3858 - accuracy: 0.8996


## Evaluate perfomance

In [26]:
test_loss, test_acc = model.evaluate(X_test, y_test)
print("Test accuracy:", test_acc)

313/313 [==============================] - 1s 2ms/step - loss: 0.3205 - accuracy: 0.9327
Test accuracy: 0.932699978351593


“Model evaluated on 313 batches of test data and achieved a test accuracy of ~93%, with a final loss of 0.32. That means the model generalized well on unseen data and didn’t overfit